services/time/examples/simple_service.rs

```diff
@@ -71,28 +73,35 @@ impl<'a> MarkerSchema<&'a mut Fork> {
 
 transactions! {
     MarkerTransactions {
-        const SERVICE_ID = SERVICE_ID;
 
         /// Transaction, which must be executed no later than the specified time (field `time`).
         struct TxMarker {
-            from: &PublicKey,
             mark: i32,
             time: DateTime<Utc>,
         }
     }
 }
 
-impl Transaction for TxMarker {
-    fn verify(&self) -> bool {
-        self.verify_signature(self.from())
+impl TxMarker {
+    fn sign(
+        mark: i32,
+        time: DateTime<Utc>,
+        pk: &PublicKey,
+        sc: &SecretKey,
+    ) -> Message<RawTransaction> {
+        Protocol::sign_transaction(TxMarker::new(mark, time), SERVICE_ID, *pk, sc)
     }
+}
 
-    fn execute(&self, view: &mut Fork) -> ExecutionResult {
+impl Transaction for TxMarker {
+    fn execute(&self, mut tc: TransactionContext) -> ExecutionResult {
```

Comment:
I will be resident @aleksuss here to ask for a rename into `context`, for the sake of consistency.

(Ditto for `TxTime`.)